### ISSUE TITLE
Delete a fail over server after all pending request have finished

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1526]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1522
- Created at 2012-09-12 19:49:16 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:12:25 as wontfix
- Assigned to nobody
---

The list of resolved servers is allocated on the back end context and kept in the fo_service structure. However, a single request often resolves a server and keeps a pointer until the end of a request and only then gives feedback about the server based on the request result.

This presents a big race condition in case the SRV resolution is used.  When there are requests coming in in parallel, it is possible that an incoming request will invalidate a server until another request that holds a pointer to the original server is able to give a feedback.

A patch in 1.9.0 RC1 simply checks if a server is in the list of servers maintained by a service before reading its status.

It would be better if the fail over kept a list of requests that are pending for a server and only actually freed the server after all the pending requests are completed. That's outside the scope of 1.9.0 RC1, so I'm filing it as a separate ticket.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-09-13 15:17:18](https://pagure.io/SSSD/sssd/issue/1522#243536)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.10.0
rhbz:  => 0
type: defect => task

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-09-17 23:30:15](https://pagure.io/SSSD/sssd/issue/1522#243537)**

Fields changed

milestone: SSSD 1.10.0 => SSSD 1.10 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1522#243538)**

Fields changed

selected:  => Not need

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-01-02 15:32:08](https://pagure.io/SSSD/sssd/issue/1522#243539)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/1522#243540)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-06-25 15:32:44](https://pagure.io/SSSD/sssd/issue/1522#243541)**

Should be done together with the failover rework.

changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
milestone: SSSD 1.13 beta => SSSD 1.14 beta
review:  => 0
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-02-16 14:06:02](https://pagure.io/SSSD/sssd/issue/1522#243542)**

Refactoring the fail over is not scoped for 1.14

milestone: SSSD 1.14 beta => SSSD 1.15 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 15:08:58](https://pagure.io/SSSD/sssd/issue/1522#243543)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-13 14:22:33](https://pagure.io/SSSD/sssd/issue/1522#633186)**

**Metadata Update from @thalman**:
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue tagged with: Canditate to close

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:12:24](https://pagure.io/SSSD/sssd/issue/1522#636694)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:12:25](https://pagure.io/SSSD/sssd/issue/1522#636695)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
